### PR TITLE
77 survey add test coverage and ci check for survey service

### PR DESCRIPTION
**Pull request by Gosqu248**
This pull request introduces several significant updates to the `survey` project, focusing on testing, code quality improvements, and build configuration. The changes include the addition of CI workflows, enhancements to the Maven build configuration, code refactoring, and the introduction of integration tests for the `SurveyService` and `SurveyController`.

### CI/CD Enhancements:
* Added a GitHub Actions workflow (`survey/.github/workflows/test.yml`) to automate backend tests, enforce a code coverage threshold of 80%, and upload JaCoCo coverage reports.

### Build Configuration:
* Updated `survey/pom.xml` to include the `maven-compiler-plugin` for annotation processing (e.g., Lombok) and the `jacoco-maven-plugin` for code coverage analysis, with a minimum coverage threshold of 80%.

### Code Refactoring:
* Refactored `SurveyService` to use a `final` `RestTemplate` and moved private helper methods (`fromSurvey`, `fromSurveyAnswers`, `toSurvey`) to the bottom of the class for improved readability. [[1]](diffhunk://#diff-b4866a03b9ed116454eb642a00eabbc7e44031e2f3533d5d3d0e4d6fc0b23880L37-R37) [[2]](diffhunk://#diff-b4866a03b9ed116454eb642a00eabbc7e44031e2f3533d5d3d0e4d6fc0b23880R79-R111)
* Simplified imports by removing unused ones in `SurveyService`.
* Added the `@Builder` annotation to the `Task` model for streamlined object creation using Lombok.

### Testing Improvements:
* Introduced integration tests for `SurveyController` (`survey/src/test/java/com/formulai/survey/integrationTests/SurveyControllerTest.java`) to validate endpoints, including scenarios for invalid input and edge cases.
* Added integration tests for `SurveyService` (`survey/src/test/java/com/formulai/survey/integrationTests/SurveyServiceTest.java`) to ensure correctness of service methods and handle various edge cases.

obraz_2025-05-13_022640481

### CODE REVIEW

**Review by rkorszun — APPROVED** [review]

File: survey/src/test/java/com/formulai/survey/integrationTests/SurveyControllerTest.java [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+                objectMapper.getTypeFactory().constructCollectionType(List.class, SurveyAnswerResponse.class));
+
+        assertNotNull(responses);
+        assertFalse(responses.isEmpty());
+        assertEquals(surveyId, responses.getFirst().surveyId());
+    }
+
+
+    @Test
+    void createSurvey_shouldReturn400WithInvalidRequest() throws Exception {
```
Comment (imatw4r) [inline survey/src/test/java/com/formulai/survey/integrationTests/SurveyControllerTest.java:106]: Nit: The test name doesn't really tell us what's the specific case that's being tested here. How is the request invalid? 
In test below you specify both, the outcome and a reason `createSurvey_shouldReturn400WithEmptyName`

File: survey/src/test/java/com/formulai/survey/integrationTests/SurveyControllerTest.java [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+
+        // when & then
+        mockMvc.perform(post("/v1/surveys")
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content(requestJson))
+                .andExpect(status().isBadRequest());
+    }
+
+    @Test
+    void getSurvey_shouldHandleInvalidUUID() throws Exception {
```
Comment (imatw4r) [inline survey/src/test/java/com/formulai/survey/integrationTests/SurveyControllerTest.java:145]: `getSurvey_shouldReturn400WhenInvalidUUIDProvided`?
